src/RoslynInsertionTool/RoslynInsertionTool/CoreXT.cs

```diff
@@ -135,14 +142,32 @@ public List<GitChange> SaveComponents()
                 .Select(x => x.Attribute("version")).SingleOrDefault();
         }
 
-        private static XAttribute? GetVersionAttributeInPropsFileOpt(XDocument document, PackageInfo packageInfo)
+        /// <summary>
+        /// Returns the appropriate element that contains the version of the specified package. Will return either a <see cref="XAttribute" /> or <see cref="XElement"/> depending.
+        /// </summary>
+        private static XObject? GetVersionAttributeOrElementInPropsFile(XDocument document, PackageInfo packageInfo)
         {
             // Support both the existing tags - "PackageVersion" and "Include" as well the legacy tags "PackageReference" and "Update" so we could still update servicing branches.
-            return document?.Root
+            var attribute = document.Root
                 .Elements().Where(e => string.Equals(e.Name.LocalName, "ItemGroup"))
                 .Elements().Where(e => string.Equals(e.Name.LocalName, "PackageVersion") || string.Equals(e.Name.LocalName, "PackageReference"))
                 .Where(p => p.Attribute("Include")?.Value == packageInfo.PackageName || p.Attribute("Update")?.Value == packageInfo.PackageName)
                 .Select(x => x.Attribute("Version")).SingleOrDefault();
+
+            if (attribute is null)
+                return null;
+
+            // If the attribute is a property substitution, then we'll see if we actually can find the property itself
+            if (attribute.Value.StartsWith("$(") && attribute.Value.EndsWith(")"))
+            {
+                var propertyName = attribute.Value.Substring(2, attribute.Value.Length - 3);
```

Comment:
are we able to use ranges in this project?

Reply:
We're targeting net46 so no idea. I'd also have to learn how to use ranges for this. :smile: